plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -0,0 +1,214 @@
+#pragma once
+
+#include <eosio/chain/producer_schedule.hpp>
+
+namespace eosio::auto_bp_peering {
+
+///
+/// This file implements the functionality for block producers automatically establishing p2p connections to their
+/// neighbors on the producer schedule.
+///
+
+using chain::account_name;
+using chain::flat_map;
+using chain::flat_set;
+
+template <typename Derived, typename Connection>
+class connection_manager {
+#ifdef BOOST_TEST_MODULE
+ public:
+#endif
+
+   flat_map<account_name, std::string> bp_peer_addresses;
+   flat_map<std::string, account_name> bp_peer_accounts;
```

Comment:
`bp_peer_addresses` and `bp_peer_accounts` are only modified on plugin_startup so they are currently thread safe to view from any threads. However, if we add ability to modify the configuration without restarting nodeos these will need some kind of thread protection. A comment about thread safety would be nice.